### ISSUE TITLE
Uppercase vs lowercase for base >10

### ISSUE THREAD

**Issue by lassik** [s-expressionists/wscl#25]
Does the standard say anywhere if hex and other base>10 numbers should be printed using uppercase (as opposed to lowercase) letters? Uppercase seems to be the norm.

**Comment by yitzchak** [comment]
Yes, the description for DIGIT-CHAR says specifically "If the resulting [character](http://clhs.lisp.se/Body/26_glo_c.htm#character) is to be an [alphabetic](http://clhs.lisp.se/Body/26_glo_a.htm#alphabetic)[1] [character](http://clhs.lisp.se/Body/26_glo_c.htm#character), it will be an uppercase [character](http://clhs.lisp.se/Body/26_glo_c.htm#character)." There maybe other places also.

**Comment by lassik** [comment]
## Uppercase

Function `DIGIT-CHAR` (http://clhs.lisp.se/Body/f_digit_.htm#digit-char)

> If the resulting [character](http://clhs.lisp.se/Body/26_glo_c.htm#character) is to be an [alphabetic](http://clhs.lisp.se/Body/26_glo_a.htm#alphabetic)[1] [character](http://clhs.lisp.se/Body/26_glo_c.htm#character), it will be an uppercase [character](http://clhs.lisp.se/Body/26_glo_c.htm#character).

## No opinion

Variable `*PRINT-BASE*`, `*PRINT-RADIX*` (http://clhs.lisp.se/Body/v_pr_bas.htm#STprint-baseST)

> For radices above 10, letters of the alphabet are used to represent digits above 9.

> If the [value](http://clhs.lisp.se/Body/26_glo_v.htm#value) of [*print-radix*](http://clhs.lisp.se/Body/v_pr_bas.htm#STprint-radixST) is [true](http://clhs.lisp.se/Body/26_glo_t.htm#true), the printer will print a [radix](http://clhs.lisp.se/Body/26_glo_r.htm#radix) specifier to indicate the radix in which it is printing a [rational](http://clhs.lisp.se/Body/26_glo_r.htm#rational) number. The radix specifier is always printed using lowercase letters. If [*print-base*](http://clhs.lisp.se/Body/v_pr_bas.htm#STprint-baseST) is 2, 8, or 16, then the radix specifier used is #b, #o, or #x, respectively.

2.3.2.1.1 Syntax of an Integer (http://clhs.lisp.se/Body/02_cbaa.htm)

13.1.4.6 Digits in a Radix (http://clhs.lisp.se/Body/13_adf.htm)

22.1.3.1.1 Printing Integers (http://clhs.lisp.se/Body/22_acaa.htm)

22.3.2.1 Tilde R: Radix (http://clhs.lisp.se/Body/22_cba.htm)

22.3.2.5 Tilde X: Hexadecimal (http://clhs.lisp.se/Body/22_cbe.htm)

_current output base_ (http://clhs.lisp.se/Body/26_glo_c.htm#current_output_base)

_digit_ (http://clhs.lisp.se/Body/26_glo_d.htm#digit)

_radix_ (http://clhs.lisp.se/Body/26_glo_r.htm#radix)

**Comment by informatimago** [comment]
Note that if you want a specific case, you can use format ~:@( or ~( 
```
cl-user> (format t "~:@(~X~) ~:*~(~X~)~%" #xbabe)
BABE babe
nil
```
